fluss-server/src/main/java/org/apache/fluss/server/coordinator/statemachine/ReplicaStateMachine.java

```diff
@@ -376,12 +380,16 @@ private void logInvalidTransition(
     }
 
     private void logFailedSateChange(
-            TableBucketReplica replica, ReplicaState currState, ReplicaState targetState) {
+            TableBucketReplica replica,
+            ReplicaState currState,
+            ReplicaState targetState,
+            Throwable reason) {
         LOG.error(
                 "Fail to change state for table bucket replica {} from {} to {}.",
                 stringifyReplica(replica),
                 currState,
-                targetState);
+                targetState,
+                reason);
```

Comment:
We previously only logs the exception message, however, the new version prints the full stack of the exception which may make the error message complex and verbose. However still keeping log error message? In this way, we can also simplify the method without creating an Exception object for it.